extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js

```diff
@@ -442,28 +465,45 @@ ve.init.mw.ViewPageTarget.prototype.loadAndRenderFancyCaptcha = function ( $cont
 	} );
 };
 
-/**
- * Get the licensing
- *
- * @returns {jQuery}
- */
-ve.init.wikia.ViewPageTarget.prototype.getLicense = function () {
-	if ( !this.$license ) {
-		this.$license = this.$('<div>')
-			.append(
-				this.$( '<p>' ).addClass( 've-ui-wikia-license' )
-					.html( ve.init.platform.getParsedMessage( 'copyrightwarning' ) )
-					.find( 'a' ).attr( 'target', '_blank' ).end()
-			);
+ve.init.wikia.ViewPageTarget.prototype.shouldReloadPageAfterSave = function () {
+	return this.userLoggedInDuringEdit ||
+		ve.init.wikia.ViewPageTarget.super.prototype.shouldReloadPageAfterSave.call( this );
+};
+
+ve.init.wikia.ViewPageTarget.prototype.updatePageOnCancel = function () {
+	if ( this.userLoggedInDuringEdit === true ) {
+		this.deactivating = true;
+		this.tearDownBeforeUnloadHandler();
+		// Reload the page so we don't need to worry about the user's state
+		window.location = this.viewUri;
+	} else {
+		ve.init.wikia.ViewPageTarget.super.prototype.updatePageOnCancel.call( this );
 	}
-	return this.$license;
 };
 
-/**
- * Set up the license, attaching it after a surface.
- *
- * @param {ve.ui.Surface} surface Surface
- */
-ve.init.wikia.ViewPageTarget.prototype.setupLicense = function ( surface ) {
-	this.getLicense().insertAfter( surface.$element.closest( '.WikiaArticle' ) );
+ve.init.wikia.ViewPageTarget.prototype.onLogInSuccess = function () {
+	new mw.Api().get( {
+		action: 'query',
+		meta: 'userinfo',
+		prop: 'info',
+		// Try to send the normalised form so that it is less likely we get extra data like
+		// data.normalised back that we don't need.
+		titles: new mw.Title( this.pageName ).toText(),
+		indexpageids: '',
+		intoken: 'edit'
+	} ).done( function ( data ) {
+		var userInfo = data.query && data.query.userinfo,
+			pageInfo = data.query && data.query.pages && data.query.pageids &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)